pkg/executor/recover_test.go

```diff
@@ -690,3 +690,37 @@ func MockGC(tk *testkit.TestKit) (string, string, string, func()) {
 	tk.MustExec("delete from mysql.tidb where variable_name in ( 'tikv_gc_safe_point','tikv_gc_enable' )")
 	return timeBeforeDrop, timeAfterDrop, safePointSQL, resetGC
 }
+
+func TestFlashbackClusterWithManyDBs(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+
+	timeBeforeDrop, _, safePointSQL, resetGC := MockGC(tk)
+	defer resetGC()
+
+	// Set GC safe point.
+	tk.MustExec(fmt.Sprintf(safePointSQL, timeBeforeDrop))
+
+	backup := kv.TxnEntrySizeLimit.Load()
+	kv.TxnEntrySizeLimit.Store(50000)
+	t.Cleanup(func() {
+		kv.TxnEntrySizeLimit.Store(backup)
+	})
+
+	tk.MustExec("set @@global.tidb_ddl_error_count_limit = 2")
+	tk.MustExec("set @@global.tidb_enable_fast_create_table=ON")
+
+	for i := 0; i < 400; i++ {
```

Comment:
you can create them in parallel to be faster, as in https://github.com/pingcap/tidb/pull/55025/files#diff-3c769a746cec11d5c583a7ad47b92b8a90a1b8d8e3cb86a7f368de7deb5f71b9R660-R661